payment.php

```diff
@@ -108,8 +108,13 @@ function generate_address($blockonomics, $ca) {
 				/*
 				 * PRICE GENERATION
 				 */
-				$btc_amount = $blockonomics->getBitcoinAmount($fiat_amount, $currency);
-				$ca->assign('btc_amount', $btc_amount / 1.0e8);
+				if($currency == 'BTC'){
+					$btc_amount = $fiat_amount;
```

Comment:
This calculation works for frontend but is incorrect for db (notice how `else` is storing different amount to template and to `$btc_amoun`). Bits column which keeps track of BTC amount to pay is an int column and expects satoshi amount, here we try to store BTC amount in a float format later on line 122 `insertOrderToDb()`. Bits ends up 0 in db